.travis.yml

```diff
@@ -1,57 +1,70 @@
 language: php
+dist: trusty
+sudo: false 
 
-php:
-  - 5.4
-  - 5.5
-  - 5.6
-  - 7.0
+services:
+  - mongodb
 
 env:
   global:
-    - KEY_SERVER="hkp://keyserver.ubuntu.com:80"
-    - MONGO_REPO_URI="http://repo.mongodb.com/apt/ubuntu"
-    - MONGO_REPO_TYPE="precise/mongodb-enterprise/"
-    - SOURCES_LOC="/etc/apt/sources.list.d/mongodb.list"
-  matrix:
-    - DRIVER_VERSION=1.2.0 SERVER_VERSION=2.6
-    - DRIVER_VERSION=1.2.0 SERVER_VERSION=3.0
-    - DRIVER_VERSION=1.2.0 SERVER_VERSION=3.2
+    - DRIVER_VERSION=stable
+
+addons:
+  apt:
+    packages: &common_packages
+      - gdb
 
 matrix:
   fast_finish: true
   include:
+    - php: 5.5
+      addons: &common_addons
+        apt: 
+          sources: [ mongodb-3.4-precise ]
+          packages: [ mongodb-org, *common_packages ]
+    - php: 5.6
+      addons: *common_addons
     - php: 7.0
-      env: DRIVER_VERSION=1.2.0 SERVER_VERSION=2.4
+      addons: *common_addons
+    - php: 7.1
+      addons: *common_addons
+    - php: 7.2
+      addons: *common_addons
     - php: 7.0
-      env: DRIVER_VERSION=devel SERVER_VERSION=3.2
-  exclude:
-    - php: 5.4
-      env: DRIVER_VERSION=stable SERVER_VERSION=2.6
-    - php: 5.4
-      env: DRIVER_VERSION=stable SERVER_VERSION=3.0
-    - php: 5.5
-      env: DRIVER_VERSION=stable SERVER_VERSION=2.6
-    - php: 5.5
-      env: DRIVER_VERSION=stable SERVER_VERSION=3.0
-
-before_install:
-  - sudo apt-key adv --keyserver ${KEY_SERVER} --recv 7F0CEB10
-  - sudo apt-key adv --keyserver ${KEY_SERVER} --recv EA312927
-  - echo "deb ${MONGO_REPO_URI} ${MONGO_REPO_TYPE}${SERVER_VERSION} multiverse" | sudo tee ${SOURCES_LOC}
-  - sudo apt-get update -qq
-
-install:
-  - if dpkg --compare-versions ${SERVER_VERSION} le "2.4"; then export SERVER_PACKAGE=mongodb-10gen-enterprise; else export SERVER_PACKAGE=mongodb-enterprise; fi
-  - sudo apt-get install ${SERVER_PACKAGE}
-  - sudo apt-get -y install gdb
+      env: DRIVER_VERSION=1.2.0
+      addons:
+        apt:
+          sources: [ mongodb-upstart ]
+          packages: [ mongodb-10gen, *common_packages ]
+    - php: 7.0
+      env: DRIVER_VERSION=1.2.0
+      addons:
+        apt:
+          sources: [ mongodb-upstart ]
+          packages: [ mongodb-org, *common_packages ]
```

Comment:
Line 38 uses `mongodb-10gen` (MongoDB 2.4), while this uses `mongodb-org` (2.6).